portlayer/exec/container.go

```diff
@@ -135,5 +138,29 @@ func (c *Container) Start(ctx context.Context) error {
 		return err
 	}
 
-	return nil
+	waitFunc := func(pc []types.PropertyChange) bool {
+		// guestinfo key that we want to wait for
+		key := fmt.Sprintf("guestinfo..sessions|%s.started", c.ID)
+
+		for _, c := range pc {
+			if c.Op != types.PropertyChangeOpAssign {
+				continue
+			}
+
+			values := c.Val.(types.ArrayOfOptionValue).OptionValue
+			for _, value := range values {
+				// check the status of the key and return true if it's "true"
+				if key == value.GetOptionValue().Key && value.GetOptionValue().Value.(string) == "true" {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	// Wait some before giving up...
+	ctx, cancel := context.WithTimeout(ctx, propertyCollectorTimeout)
+	defer cancel()
+
+	return c.vm.WaitForExtraConfig(ctx, waitFunc)
```

Comment:
I don't know how error return from Start gets handled at this point, but if we fail to launch we should be returning an error, so return value is based around `value == "true"`